src/modules/EventModule.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include <qtimer.h>
+#include "core/Module.h"
+#include "UmikoBot.h"
+
+class EventModule : public Module
+{
+public:
+	struct RaffleDraw
+	{
+	public:
+		snowflake_t m_UserId;
+		QList<unsigned int> m_TicketIds;
+		RaffleDraw(snowflake_t userID)
+			:m_UserId(userID){}
+	};
+
+	struct EventConfig
+	{
+		QTimer* eventTimer{ nullptr };
+		QTimer* raffleDrawRewardClaimTimer{ nullptr };
+		bool isEventRunning{ false };
```

Comment:
Nope, its needed because it activates whenever an event is running that can be any event. It tracks the event system is in use or not. The event timer is based on it. 🙂